File: fixtures/scale/python/large/cipherlab/checks/check_splitpack.py
from cipherlab.ciphers.splitpack import splitpack_encode, splitpack_decode
from cipherlab.specs.splitpack_spec import splitpack_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_splitpack():
    spec = splitpack_spec()
    for text in corpus_codec():
        packed = splitpack_encode(text)
        if splitpack_decode(packed) != text:
            return False
    return spec["category"] == "codec"

File: fixtures/scale/ruby/large/lib/check_splitpack.rb
require_relative 'splitpack_encode'
require_relative 'splitpack_decode'
require_relative 'corpus_codec'

def check_splitpack
  corpus_codec.each do |text|
    packed = splitpack_encode(text)
    return false if splitpack_decode(packed) != text
  end
  true
end
